pallets/fiat_filter/src/lib.rs

```diff
@@ -0,0 +1,218 @@
+        + IsSubType<did::Call<Self>>
+        + IsSubType<anchor::Call<Self>>
+        + IsSubType<blob::Call<Self>>
+        + IsSubType<revoke::Call<Self>>
+        + IsSubType<attest::Call<Self>>;
+}
+
+decl_error! {
+    pub enum Error for Module<T: Config> where BalanceOf<T>: From<u64> {
+        /// Failed calculation because of numeric type overflow
+        ArithmeticOverflow,
+        /// checked_div: tried to divide by zero
+        DivideByZero,
+        /// Call is not among the core_mods ones that should go through fiat_filter
+        UnexpectedCall,
+        /// Call to T::PriceProvider::optimized_get_dock_usd_rate() failed
+        NoPriceFound,
+    }
+}
+
+// The module's callable functions (Dispatchable functions)
+// Dispatchable functions must be annotated with a weight and must return a DispatchResult.
+decl_module! {
+    pub struct Module<T: Config> for enum Call
+    where
+        origin: T::Origin,
+        BalanceOf<T>: From<u64>,
+    {
+        // Errors must be initialized if they are used by the pallet.
+        type Error = Error<T>;
+
+        /// Execute a Call. Must be a core_mods call (DID, Blob, Anchor, Revoke/Unrevoke, Attest)
+        #[weight = 10_000]
+        pub fn execute_call(origin, call: Box<<T as Config>::Call>) -> DispatchResultWithPostInfo {
+            Ok(Self::execute_call_(origin, &call)?)
+        }
+    }
+}
+
+type BalanceOf<T> =
+    <<T as Config>::Currency as Currency<<T as system::Config>::AccountId>>::Balance;
+
+type AmountUsd = u64;
+
+// all prices given in nUSD (billionth USD)
+pub const PRICE_DID_OP: u64 = 100_000; // 100_000/1B or 0.0001 USD
+pub const PRICE_ANCHOR_OP_PER_BYTE: u64 = 1000;
+pub const PRICE_BLOB_OP_PER_BYTE: u64 = 1000;
+pub const PRICE_REVOKE_REGISTRY_OP: u64 = 40_000;
+pub const PRICE_REVOKE_REVOCATION_OP: u64 = 20_000;
+pub const PRICE_ATTEST_OP_PER_BYTE: u64 = 1000;
+
+// private helper functions
+impl<T: Config> Module<T>
+where
+    BalanceOf<T>: From<u64>,
+{
+    /// Get fee in fiat unit for a given Call
+    /// Result expressed in nUSD (billionth USD)
+    fn get_call_fee_fiat_(
+        call: &<T as Config>::Call,
+    ) -> Result<AmountUsd, DispatchErrorWithPostInfo> {
+        // TODO make sure we discuss and decide the actual pricing for each call type
+        match call.is_sub_type() {
+            Some(did::Call::new(_did, _detail)) => return Ok(PRICE_DID_OP), // 50/1B or 0.00000005 USD
+            Some(did::Call::update_key(_key_update, _sig)) => return Ok(PRICE_DID_OP),
+            Some(did::Call::remove(_to_remove, _sig)) => return Ok(PRICE_DID_OP),
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(anchor::Call::deploy(bytes)) => {
+                return Ok(PRICE_ANCHOR_OP_PER_BYTE * bytes.len() as u64)
+            }
+            _ => {}
+        };
+        match call.is_sub_type() {
+            Some(blob::Call::new(blob, _sig)) => {
+                let size: u64 = 64 + blob.blob.len() as u64;
```

Comment:
What is 64 for?

Reply:
other bytes in the blob. But I removed that and made it depend only on the size of the blob itself.